src/search_procedure.jl

```diff
@@ -16,4 +16,5 @@ function search(g::Grammars.Grammar, problem::Data.Problem, depth::Int, enumerat
             return expr
         end
     end
+    return nothing
```

Comment:
what about returning the best program found so far instead of `nothing`?
If we cannot have the best one, something is better than nothing

Reply:
Good idea!
Assuming we define the best program found so far as the program that works on the most examples, this will require a change in the evaluator. 
Given that this PR is already linked with 3 PRs in other modules, I think we should turn this into a separate issue.

Reply:
We do have to think of a way to distinguish between optimal and suboptimal solutions though

Reply:
agree with putting it as a separate issue.
Distinguishing between optimal and suboptimal (do you mean a program that solves all examples from the one that solves some?) program is not that difficult? It can be also done later: when you get a solution, test whether it is optimal or not. Or we can return a boolean together with a program?

Reply:
I'd vote for returning the boolean

Reply:
that sounds good to me. Let's go with it.